src/Apps/Auction/Routes/Register/index.tsx

```diff
@@ -13,27 +7,69 @@ import {
   graphql,
   RelayProp,
 } from "react-relay"
+import { TrackingProp } from "react-tracking"
 import { data as sd } from "sharify"
+
+import { Register_me } from "__generated__/Register_me.graphql"
+import { Register_sale } from "__generated__/Register_sale.graphql"
+import {
+  RegisterCreateBidderMutation,
+  RegisterCreateBidderMutationResponse,
+} from "__generated__/RegisterCreateBidderMutation.graphql"
+import { RegisterCreateCreditCardMutation } from "__generated__/RegisterCreateCreditCardMutation.graphql"
+
+import {
+  FormValues,
+  StripeWrappedRegistrationForm,
+} from "Apps/Auction/Components/RegistrationForm"
+import { AppContainer } from "Apps/Components/AppContainer"
+import { trackPageViewWrapper } from "Apps/Order/Utils/trackPageViewWrapper"
+import { track } from "Artsy"
+import * as Schema from "Artsy/Analytics/Schema"
+import { ErrorModal } from "Components/Modal/ErrorModal"
 import createLogger from "Utils/logger"
-import { StripeWrappedRegistrationForm } from "../../Components/RegistrationForm"
-import { FormValues } from "../../Components/RegistrationForm"
 
 const logger = createLogger("Apps/Auction/Routes/Register")
 
 interface RegisterProps {
   sale: Register_sale
-  relay?: RelayProp
+  me: Register_me
+  relay: RelayProp
+  tracking: TrackingProp
 }
 
 export const RegisterRoute: React.FC<RegisterProps> = props => {
-  const { relay, sale } = props
+  const { me, relay, sale, tracking } = props
   const [showErrorModal, setShowErrorModal] = useState(false)
 
+  const commonProperties = {
+    auction_slug: sale.id,
+    auction_state: sale.status,
+    sale_id: sale._id,
+    user_id: me.id,
+  }
+
+  function trackRegistrationFailed(errors: string[]) {
+    tracking.trackEvent({
+      action_type: Schema.ActionType.RegistrationSubmitFailed,
+      error_messages: errors,
+      ...commonProperties,
+    })
+  }
+
+  function trackRegistrationSuccess(bidderId: string) {
+    tracking.trackEvent({
+      action_type: Schema.ActionType.RegistrationSubmitted,
+      bidderId,
+      ...commonProperties,
+    })
+  }
+
   function createBidder() {
     return new Promise(async (resolve, reject) => {
       commitMutation<RegisterCreateBidderMutation>(relay.environment, {
-        onCompleted: (data, errors) => {
-          resolve()
+        onCompleted: data => {
```

Comment:
Do we need to be handling the `(data, errors)` case rather than resolving unconditionally? Where does this second errors parameter come from rather than calling the `onError` parameter below?

Reply:
`errors` here is different than `onError`. The `errors` parameter here are any potential errors that are returned from metaphysics. 

`onError` is an error that occurred with the request. I.E. the call to metaphysics likely failed.